### DISCUSSION TITLE
Is it possible to programmatically log into OpenMPF?

### DISCUSSION THREAD

**Discussion by zramsis** [openmpf/openmpf#1609] [Q&A] [answered]
There doesn't seem to be a log in api call available in the rest api. I wanted to check if there was any other way.

**Comment by jrobble** [comment]
There's no login endpoint, but you should be able to provide basic auth credentials when using any of the exposed REST endpoints. For example:

```
curl -X GET -u "admin:mpfadm" "http://somehost:8080/workflow-manager/rest/jobs/1" -H  "accept: */*"
```

**Reply by zramsis** [reply]
So for my project, we are creating a separate app that will run alongside OpenMPF. On our app, we are hoping to display the media that is stored in OpenMPF. Right now we have our app running on port 8081 and OpenMPF running on 8080 and we display the media using a uri to the path of the image like:
http://localhost:8080/workflow-manager/server/node-image?nodeFullPath=%2Fopt%2Fmpf%2Fshare%2Fremote-media%2Fimage.jpg

Now this works.. but only when we are logged into OpenMPF. So that is why I am looking for a way to log in directly. Is there another method you might suggest to get our desired functionality of displaying the media?

**Reply by jrobble** [reply, marked as answer]
Sorry for the late reply. I didn't receive or notice an email notification for this discussion.

Unfortunately we don't expose the media server endpoints as part of our public API. If you like, you can update the [ServerMediaController code](https://github.com/openmpf/openmpf/blob/master/trunk/workflow-manager/src/main/java/org/mitre/mpf/mvc/controller/ServerMediaController.java) to expose them. Please refer to [this answer](https://github.com/openmpf/openmpf/discussions/1573#discussioncomment-3416917) for more details on exposing endpoints.
